src/rpc-client.ts

```diff
@@ -1,95 +1,263 @@
+        const response = await this._client.blockchain.getAccountByAddress(address);
+        return Number(response.data.balance);
     }
 
-    async getTransactionReceipt(txHash) {
-        return this._jsonRpcFetch('getTransactionReceipt', txHash);
+    /**
+     * Gets transaction details by hash
+     * @param txHash - The transaction hash
+     * @returns Promise resolving to transaction details
+     * @throws If transaction not found or hash invalid
+     */
+    async getTransactionReceipt(txHash: string): Promise<Transaction> {
+        if (!txHash || typeof txHash !== 'string') {
+            throw new Error('Invalid transaction hash');
+        }
+        const response = await this._client.blockchain.getTransactionByHash(txHash);
+        if (!response.data) {
+            throw new Error('Transaction not found');
+        }
+        return response.data as Transaction;
     }
 
-    async getTransactionsByAddress(address) {
-        return this._jsonRpcFetch('getTransactionsByAddress', address.toUserFriendlyAddress());
+    /**
+     * Sends a transaction
+     * @param params - Transaction parameters
+     * @returns Promise resolving to transaction result
+     */
+    async sendTransaction(params: TransactionParams) {
+        
+        try {
+            const response = await this._client.consensus.sendTransaction({
+                ...params,
+                relativeValidityStartHeight: 0
+            });
+            if (!response.data) throw new Error('Failed to send transaction');
+            return response.data;
+        } catch (error) {
+            console.error('RPC Error details:', error);
+            throw new Error(`Failed to send transaction: ${error instanceof Error ? error.message : 'Unknown error'}`);
+        }
     }
 
-    async sendTransaction(transaction) {
-        return this._jsonRpcFetch('sendRawTransaction', BufferUtils.toHex(transaction.serialize()));
+    /**
+     * Sends a raw transaction using direct transaction creation and signing
+     * @param params - Raw transaction parameters including KeyPair for signing
+     * @returns Promise resolving to transaction result
+     */
+    async sendRawTransaction(params: RawTransactionParams) {
+        try {
+            const blockHeight = await this.getBlockHeight();
+            // Create transaction using TransactionBuilder
+            const transaction = TransactionBuilder.newBasic(
+                params.sender.publicKey.toAddress(),
+                Address.fromUserFriendlyAddress(params.recipient),
+                BigInt(params.value),
+                BigInt(params.fee || 0),
+                blockHeight,
+                5
+            );
+
+            // Sign the transaction
+            transaction.sign(params.sender);
+
+            const data = await this._client.call<any>({ method: 'sendRawTransaction', params: [ BufferUtils.toHex(transaction.serialize())] });
+            return data.data;
```

Comment:
Nicer with destructuring. Similarly at other places.
```suggestion
            const { data } = await this._client.call<any>({ method: 'sendRawTransaction', params: [ BufferUtils.toHex(transaction.serialize())] });
            return data;
```